src/locales/pt.ts

```diff
@@ -5,41 +5,49 @@ import { printValue, LocaleObject } from 'yup';
 // Based on https://github.com/jquense/yup/blob/b940eef48eb7456622ae384d0ffa7363d4fbad25/src/locale.ts
 export const mixed: LocaleObject['mixed'] = {
   default: '${path} é inválido.',
-  required: '${path} é um campo necessário',
-  defined: '${path} deve ser definido',
-  notNull: '${path} não pode ser nulo',
-  oneOf: '${path} deve ser um dos seguintes valores: ${values}',
-  notOneOf: '${path} não deve ser um dos seguintes valores: ${values}',
+  required: '${path} é obrigatório',
+  defined: '${path} não deve ser indefinido',
+  notNull: '${path} não pode ser vazio',
+  oneOf: '${path} deve ter um dos seguintes valores: ${values}',
+  notOneOf: '${path} não deve ter nenhum dos seguintes valores: ${values}',
   notType: ({ path, type, value, originalValue }) => {
     const isCast = originalValue != null && originalValue !== value;
     let msg =
-      `${path} deve ser um \`${type}, ` +
-      `Mas o valor final foi: \`${printValue(value, true)}\`` +
+      `${path} deve ser do tipo \`${type}\`, ` +
+      `mas o valor final foi: \`${printValue(value, true)}\`` +
       (isCast
-        ? ` (lançado do valor \`${printValue(originalValue, true)}\`).`
+        ? ` (cast do valor \`${printValue(originalValue, true)}\`).`
         : '.');
 
     if (value === null) {
       msg +=
-        `\n Se "nulo" for destinado a um valor vazio, marque o esquema como` +
-        ' `.nullable()`';
+        '\n Se a intenção era usar "null" como um valor em branco marque o esquema como `.nullable()`';
     }
 
     return msg;
   },
 };
 
 export const string: LocaleObject['string'] = {
-  length: '${path} deve ser exatamente ${length} caracteres',
-  min: '${path} deve ser pelo menos ${min} caracteres',
-  max: '${path} deve ser no máximo ${max} caracteres',
-  matches: '${path} deve corresponder ao seguinte: "${regex}"',
-  email: '${path} deve ser um e -mail válido',
-  url: '${path} deve ser um URL válido',
+  length: ({ path, length }) =>
+    `${path} deve ter exatamente ${length} ${
+      length === 1 ? 'caractere' : 'caracteres'
+    }`,
+  min: ({ path, min }) =>
+    `${path} deve ter no mínimo ${min} ${
+      min === 1 ? 'caractere' : 'caracteres'
+    }`,
+  max: ({ path, max }) =>
+    `${path} deve ter no máximo ${max} ${
+      max === 1 ? 'caractere' : 'caracteres'
+    }`,
+  matches: '${path} deve corresponder ao padrão: "${regex}"',
+  email: '${path} deve ser um e-mail válido',
+  url: '${path} deve ser uma URL válida',
   uuid: '${path} deve ser um UUID válido',
-  trim: '${path} deve ser uma corda aparada',
-  lowercase: '${path} deve ser uma corda minúscula',
```

Comment:
String is something that has a bad translation to pt-BR. `corda` is the same as rope in portuguese.